### ISSUE TITLE
Can't use slab::Slab with mio::Token (compilation error)

### ISSUE THREAD

**Issue by sbstp** [carllerche/mio#390]
Code: 
```rust
struct ... {
  conns: Slab<Connection, Token>,
}
```
The error I get:
```
error: the trait `slab::Index` is not implemented for the type `mio::token::Token`
```

Versions: rustc 1.8.0 (db2939409 2016-04-11) slab: 0.2.0, mio: 0.5.1

**Comment by sbstp** [comment]
Seems like this might be caused by the slab version used in mio, which is `slab = "0.1.0"`

**Comment by tikue** [comment]
Hit this too.

**Comment by rrichardson** [comment]
Thanks for posting this.  Can you provide a bit more code, or relevant Cargo.toml ?

I have verified that the tests to create Slabs with Tokens, and they compile and run on the branches  master and v0.5.x.   IOW,  I'm not sure how to repro this.

**Comment by sbstp** [comment]
This fails to compile on my machine. The way I've fixed it, for now, is just to use `slab = "0.1"`.

Cargo.toml
```toml
[package]
name = "test"
version = "0.1.0"

[dependencies]
mio = "0.5"
slab = "0.2"

```
main.rs
```rust
extern crate mio;
extern crate slab;
fn main() {
    let slab: slab::Slab<usize, mio::Token> = slab::Slab::new(1337);
}
```

**Comment by carllerche** [comment]
Yeah, mio v0.5 requires slab v0.1. If you want to use slab v0.2, you can use mio master branch. Hope this helps.
